src/opentime/rationalTime.h

```diff
@@ -16,46 +16,84 @@ enum IsDropFrameRate : int {
     
 class RationalTime {
 public:
-    explicit RationalTime(double value = 0, double rate = 1)
-    : _value {value}, _rate {rate} {}
+    explicit RationalTime(double value = 0, double rate = 1) : _value {value}, _rate {rate} {}
+    RationalTime(RationalTime && rh) { _value = rh._value; _rate = rh._rate; }
+    RationalTime(RationalTime const& rh) { _value = rh._value; _rate = rh._rate; }
+    RationalTime& operator= (RationalTime const& rh) { _value = rh._value; _rate = rh._rate; return *this; }
+    RationalTime& operator= (RationalTime && rh) { _value = rh._value; _rate = rh._rate; return *this; }
+    ~RationalTime() = default;
+
+    bool is_invalid_time() const { return _rate <= 0; }
+    double value() const { return _value; }
+    double rate() const { return _rate; }
+    
+    void set_value(double v) { _value = v; }
+    void set_rate(double r) { _rate = r; }
+    
+    RationalTime floor() const {
+        return RationalTime { std::floor(_value), _rate };
+    }
+
+    RationalTime floor(double rate) const {
+        return rescaled_to(rate).floor();
+    }
+    
+    RationalTime ceil() const {
+        return RationalTime { std::ceil(_value), _rate };
+    }
 
-    RationalTime(RationalTime const&) = default;
-    RationalTime& operator= (RationalTime const&) = default;
+    RationalTime ceil(double rate) const {
+        return rescaled_to(rate).ceil();
+    }
 
-    bool is_invalid_time() const {
-        return _rate <= 0;
+    RationalTime round_nearest() const {
+        return RationalTime(_value + 0.5 / _rate, _rate).floor();
     }
     
-    double value() const {
-        return _value;
+    RationalTime round_nearest(double rate) const {
+        return rescaled_to(rate).round_nearest();
+    }
+    
+    RationalTime incr() const {
+        RationalTime result = floor();
+        return RationalTime(result._value + 1, result._rate);
+    }
+
+    RationalTime decr() const {
+        RationalTime result = floor();
+        return RationalTime(result._value - 1, result._rate);
     }
 
-    double rate() const {
-        return _rate;
+    RationalTime incr(double rate) const {
+        return rescaled_to(rate).incr();
+    }
+
+    RationalTime decr(double rate) const {
+        return rescaled_to(rate).decr();
```

Comment:
That's correct, plus or minus one frame. They're called incr and decr because that's what they were called in the QuickTime API.

Same argument about shenanigan risk if I don't include "natural" variants. Again, I think this can be addressed via nice naming. How about

```cpp
RationalTime next_frame() const; // natural next frame
RationalTime next_frame(double rate) const; // next frame with imposed framerate
RationalTime previous_frame() const; // previous next frame
RationalTime previous_frame(double rate) const; // previous frame with imposed framerate
```

Reply:
I definitely like the explicitness of this proposed naming. My only point of contention would be that it assumes the time is referring to frames. The ergonomics are a little lower, but I like to use the term "sample" to keep the terminology media type agnostic.

Are names like `next_sample_time` and `previous_sample_time` too awful?

Reply:
As in the previous comment, my current preference is not to rename things as we haven't come up with any nomenclature less ambiguous. than where we started.